### ISSUE TITLE
tiffave rgb image with jpeg compression result false colour

### ISSUE THREAD

**Issue by heman1-test** [libvips/libvips#3743]
Hello,

I try to generate a pyramid tiff. All other compression types (deflate lzw) are OK but jpeg give me wrong colour. Input is an RGB image.
Any idea why this happens? I attached sample for testing / original and converted.

**vips tiffsave source_image output_image.tif --tile --pyramid --compression jpeg --Q=90 --tile-width 256 --tile-height 256**

[test_tif.zip](https://github.com/libvips/libvips/files/13242832/test_tif.zip)

Thank You

**Comment by jcupitt** [comment]
Hi @heman1-test,

It seems to be working for me:

```
$ unzip test_tif.zip 
Archive:  test_tif.zip
  inflating: test.tif                
  inflating: test_tiled.tiff         
$ vips copy test.tif x.tif[compression=jpeg,tile,pyramid]
$ vips copy x.tif x.jpg
$
```

To make:

![x](https://github.com/libvips/libvips/assets/580843/e3ed9821-693b-4224-b81a-8e2fa0d12edb)

Which matches `test.tif` in colour.

Maybe you have a very old libvips?

**Comment by heman1-test** [comment]
@jcupitt 
I'm using [v8.15.0-rc1](https://github.com/libvips/libvips/releases/tag/v8.15.0-rc1)
What do you use?

Ha I found it... what a bug 👎  

If you set quality to 90 and above the colour is NOK below 90 it's fine.

this one produces wrong colour **vips copy test.tif x.tif[compression=jpeg,Q=90,tile,pyramid]**

this one correct colour **vips copy test.tif x.tif[compression=jpeg,Q=89,tile,pyramid]**

**Comment by jcupitt** [comment]
Hi again @heman1-test, I agree, this looks like a bug in the new direct-mode tiff save system. I'll see if I can fix it.

**Comment by jcupitt** [comment]
OK, fixed in git head 8.15, this will b e released in 8.15.1.

Thank you very much for reporting this error @heman1-test ! I've credited you in the changelog, I hope that's OK.
